### PR TITLE
Change int64 json encoding to be string

### PR DESCRIPTION
**Pull request by TeBoring**

### CODE REVIEW

File: upb/json/printer.c [hunk 1]
```diff
@@ -208,18 +208,30 @@ static size_t fmt_bool(bool val, char* buf, size_t length) {
...
 
-static size_t fmt_int64(long val, char* buf, size_t length) {
+static size_t fmt_int32(long val, char* buf, size_t length) {
   size_t n = _upb_snprintf(buf, length, "%ld", val);
   CHKLENGTH(n > 0 && n < length);
   return n;
 }
 
-static size_t fmt_uint64(unsigned long long val, char* buf, size_t length) {
+static size_t fmt_uint32(unsigned long long val, char* buf, size_t length) {
```
Comment (haberman) [inline upb/json/printer.c]: I think uint32 should use `unsigned int`.

File: upb/json/printer.c [hunk 2]
```diff
@@ -208,18 +208,30 @@ static size_t fmt_bool(bool val, char* buf, size_t length) {
   return n;
 }
 
-static size_t fmt_int64(long val, char* buf, size_t length) {
+static size_t fmt_int32(long val, char* buf, size_t length) {
```
Comment (haberman) [inline upb/json/printer.c]: I think int32 should use `int`.

File: upb/json/printer.c [hunk 3]
```diff
@@ -208,18 +208,30 @@ static size_t fmt_bool(bool val, char* buf, size_t length) {
...
 }
 
-static size_t fmt_uint64(unsigned long long val, char* buf, size_t length) {
+static size_t fmt_uint32(unsigned long long val, char* buf, size_t length) {
   size_t n = _upb_snprintf(buf, length, "%llu", val);
   CHKLENGTH(n > 0 && n < length);
   return n;
 }
 
+static size_t fmt_int64(long val, char* buf, size_t length) {
```
Comment (haberman) [inline upb/json/printer.c]: Int64 should be `long long` (on 32-bit `long` will only be 32 bits).

File: upb/json/printer.c (hunk 1 above)
Comment (TeBoring) [reply upb/json/printer.c]: fmt_uint32 is also used for uint64 for map key

File: upb/json/printer.c (hunk 2 above)
Comment (TeBoring) [reply upb/json/printer.c]: fmt_int32 is also used for int64 for map key

File: upb/json/printer.c (hunk 3 above)
Comment (TeBoring) [reply upb/json/printer.c]: done

**Comment by TeBoring** [comment]
ERROR: /Volumes/BuildData/tmpfs/src/github/upb/bazel/upb_proto_library.bzl:191:25: name 'ProtoInfo' is not defined (did you mean 'proto_info'?)
I don't know why test failed.
bazel is updated?

**Pull request merged**